notebooks/mnist.ipynb

```diff
@@ -0,0 +1,113 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import torch\n",
+    "import matplotlib.pyplot as plt\n",
+    "from torch.utils.data import TensorDataset, DataLoader, Dataset\n",
+    "from torchvision import datasets, transforms\n",
+    "from scipy.stats import norm, chi2\n",
+    "from pqm import pqm_chi2, pqm_pvalue\n",
+    "\n",
+    "import matplotlib as mpl\n",
+    "mpl.rcParams['figure.dpi'] = 200\n",
+    "# Set font to Computer \n",
+    "# mpl.rcParams.update(mpl.rcParamsDefault)\n",
+    "plt.rc(\"font\", **{\"family\": \"serif\", \"serif\": [\"Computer Modern\"]})\n",
+    "plt.rc(\"text\", usetex=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_data = datasets.MNIST(root='../data', train=False, download=True, transform=None)\n",
+    "train_data = datasets.MNIST(root='../data', train=True, download=True, transform=None)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_data = test_data.data.reshape(-1, 28*28)/255\n",
+    "train_data = train_data.data.reshape(-1, 28*28)/255\n",
+    "all_data = torch.cat([test_data, train_data], dim=0).detach().numpy()\n",
+    "np.random.shuffle(all_data)\n",
+    "x_samples = all_data[:35000]\n",
+    "y_samples = all_data[35000:]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "zs = []\n",
+    "num_repeats = 35\n",
+    "n = len(x_samples) // num_repeats\n",
+    "\n",
+    "for i in range(num_repeats):\n",
```

Comment:
This whole block should just be `zs = pqm_chi2(x_samples, y_samples, bootstrap = 35)`